NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -34,78 +45,206 @@ function Base.show(io::IO, s::CategoryProduct)
   return print(io, ")")
 end
 
-category_show(io::IO, k, v) = print(io, v)
-
+category_show(io::IO, ::Int, v) = print(io, v)
 category_show(io::IO, k::Symbol, v) = print(io, "($k=$v,)")
 
+function Base.isless(s1::C, s2::C) where {C<:CategoryProduct}
+  return categories_isless(categories(s1), categories(s2))
+end
+function Base.isless(s1::CategoryProduct, s2::CategoryProduct)
+  return categories_isless(categories(s1), categories(s2))
+end
+
+# =======================================  shared  =========================================
+# there are 2 implementations for CategoryProduct
+# - ordered-like with a Tuple
+# - dictionary-like with a NamedTuple
+
+function categories_fusion_rule(cats1, cats2)
+  diff_cat = CategoryProduct(find_diff(cats1, cats2))
+  shared1, shared2 = find_common(cats1, cats2)
+  fused = map(fusion_rule, values(shared1), values(shared2))
+  return recover_key(typeof(shared1), fused) × diff_cat
+end
+
+# get clean results when mixing implementations
+categories_isequal(::Tuple, ::NamedTuple) = false
+categories_isequal(::NamedTuple, ::Tuple) = false
+
+categories_isless(::NamedTuple, ::Tuple) = throw(ArgumentError("Not implemented"))
+categories_isless(::Tuple, ::NamedTuple) = throw(ArgumentError("Not implemented"))
+
+categories_type(::Type{<:CategoryProduct{T}}) where {T} = T
+
+recover_key(T::Type, t::Tuple{Vararg{<:AbstractCategory}}) = sector(T, t)
```

Comment:
```suggestion
recover_key(T::Type, t::Tuple{Vararg{AbstractCategory}}) = sector(T, t)
```
I believe that will fix the warning:
```julia
┌ NDTensors
│  WARNING: Wrapping `Vararg` directly in UnionAll is deprecated (wrap the tuple instead).
│  You may need to write `f(x::Vararg{T})` rather than `f(x::Vararg{<:T})` or `f(x::Vararg{T}) where T` instead of `f(x::Vararg{T} where T)`.
└  
```